### ISSUE TITLE
Build fails with new `R-BH`: `timer.hpp: error: #error This header is deprecated and will be removed`

### ISSUE THREAD

**Issue by barracuda156** [airoldilab/sgd#111]
```
In file included from /opt/local/Library/Frameworks/R.framework/Versions/4.3/Resources/library/BH/include/boost/math/tools/roots.hpp:19,
                 from basedef.h:9,
                 from sgd.cpp:1:
/opt/local/Library/Frameworks/R.framework/Versions/4.3/Resources/library/BH/include/boost/math/tools/config.hpp:23:6: warning: #warning "The minimum language standard to use Boost.Math will be C++14 starting in July 2023 (Boost 1.82 release)" [-Wcpp]
   23 | #    warning "The minimum language standard to use Boost.Math will be C++14 starting in July 2023 (Boost 1.82 release)"
      |      ^~~~~~~
In file included from basedef.h:11:
/opt/local/Library/Frameworks/R.framework/Versions/4.3/Resources/library/BH/include/boost/timer.hpp:21:3: error: #error This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
   21 | # error This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
      |   ^~~~~
```

**Comment by barracuda156** [comment]
Apparently is fixed in `R-sgd` 1.1.2.

**Comment by jlylekim** [comment]
Thank you for the pointer--it is indeed fixed in version 1.1.2.
